Src/DataTypes/Data.cs

```diff
@@ -12,8 +13,21 @@ public enum DataType
 
     public abstract class Data
     {
+
         public DataType type;
```

Comment:
should't path and data type be readonly?
{ public get; protected set;}
or mybe even private set